File: SkillSurgeUI/src/app/core/models/interfaces/Product.ts
import { Category } from "./Category";
import { SubCategory } from "./SubCategory";
import { User } from "./User";

export interface Product {
    id: string;
    name: string;
    description?: string;
    price: number;
    currency: string;
    stockQuantity: number;
    isActive: boolean;
    createdAt: Date;
    updatedAt?: Date;
    categoryId: string;
    category?: Category;
    subCategoryId: string;
    subCategory?: SubCategory;
    userId: string;
    user?: User;
}

File: Domain/Entities/Product.cs
namespace Domain.Entities;

public class Product
{
    public Guid Id { get; set; }
    public string Name { get; set; } = default!;
    public string Description { get; set; } = default!;
    public decimal Price { get; set; }
    public int StockQuantity { get; set; }
    public bool IsActive { get; set; }
    public string Currency { get; set; } = default!;
    public Guid CategoryId { get; set; }
    public Guid? ParentCategoryId { get; set; }
    public Guid OwnerId { get; set; }
    public bool IsDeleted { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}
